test/test_math_ops.cpp

```diff
@@ -513,18 +514,100 @@ TEST(MathOps, BatchMatMul2D) {
   opexecuter.RunTest();
 }
 
+// BatchMatMul2D
+// AdjX = true
+// AdjY = false
+TEST(MathOps, BatchMatMul2DAdjX) {
+  Scope root = Scope::NewRootScope();
+
+  Tensor A(DT_FLOAT, TensorShape({2, 3}));
+  Tensor B(DT_FLOAT, TensorShape({2, 4}));
+
+  auto attrs_x = ops::BatchMatMul::Attrs();
+  attrs_x = attrs_x.AdjX(true);
+
+  AssignInputValues(A, 2.0f);
+  AssignInputValues(B, 7.0f);
+
+  vector<int> static_input_indexes = {};
+
+  auto R = ops::BatchMatMul(root, A, B, attrs_x);
+
+  vector<DataType> output_datatypes = {DT_FLOAT};
+
+  std::vector<Output> sess_run_fetchoutputs = {R};
+  OpExecuter opexecuter(root, "BatchMatMul", static_input_indexes,
+                        output_datatypes, sess_run_fetchoutputs);
+
+  opexecuter.RunTest();
+}
+
+// BatchMatMul2D
+// AdjX = false
+// AdjY = true
+TEST(MathOps, BatchMatMul2DAdjY) {
+  Scope root = Scope::NewRootScope();
+
+  Tensor A(DT_FLOAT, TensorShape({2, 4}));
+  Tensor B(DT_FLOAT, TensorShape({3, 4}));
+
+  auto attrs_y = ops::BatchMatMul::Attrs();
+  attrs_y = attrs_y.AdjY(true);
+
+  AssignInputValues(A, 2.0f);
+  AssignInputValues(B, 7.0f);
+
+  vector<int> static_input_indexes = {};
+  auto R = ops::BatchMatMul(root, A, B, attrs_y);
+
+  vector<DataType> output_datatypes = {DT_FLOAT};
+
+  std::vector<Output> sess_run_fetchoutputs = {R};
+  OpExecuter opexecuter(root, "BatchMatMul", static_input_indexes,
+                        output_datatypes, sess_run_fetchoutputs);
+
+  opexecuter.RunTest();
+}
+
+// BatchMatMul2D
+// AdjX = true
+// AdjY = true
+TEST(MathOps, BatchMatMul2DAdjXY) {
+  Scope root = Scope::NewRootScope();
+
+  Tensor A(DT_FLOAT, TensorShape({2, 3}));
+  Tensor B(DT_FLOAT, TensorShape({3, 2}));
```

Comment:
since adj is true for both, what we are actually doing here is 3x2 times 2x3
the number `2` matches for both, `3` does not need to be same in the 2 matrices. so lets make the sizes: `2x4` and `3x2` say

Reply:
Thanks :)